Earthly need to warn that buildkit daemon is down instead of SEGFAULT

the solution of the SEGFAULT was to follow this docker guide: https://docs.docker.com/engine/install/linux-postinstall/

logs:
```
$ earthly --version
earthly version v0.5.24 linux/amd64; Ubuntu 21.10

$ earthly
auto frontend could not be initialized, using stub: failed to autodetect a supported frontend
NAME:
   earthly - A build automation tool for the container era

USAGE:
     earthly [options] <target-ref>

     earthly [options] --image <target-ref>

     earthly [options] --artifact <target-ref>/<artifact-path> [<dest-path>]

     earthly [options] command [command options]

Executes Earthly builds. For more information see https://docs.earthly.dev/earthly-command.
To get started with using Earthly, check out the getting started guide at https://docs.earthly.dev/guides/basics.

VERSION:
   v0.5.24 linux/amd64; Ubuntu 21.10
...
Error: no target reference provided. Try earthly +<target-name>

$ earthly +format
auto frontend could not be initialized, using stub: failed to autodetect a supported frontend
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0xf5a6a6]

goroutine 1 [running]:
github.com/earthly/earthly/buildkitd.IsStarted(0x1374048, 0xc0002fe840, 0x12127f0, 0x11, 0x1384768, 0x1aefc70, 0x0, 0x0, 0x0)
	/earthly/buildkitd/buildkitd.go:460 +0x106
github.com/earthly/earthly/buildkitd.MaybeStart(0x1374048, 0xc0002fe840, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1aef9f8, ...)
	/earthly/buildkitd/buildkitd.go:145 +0x85
github.com/earthly/earthly/buildkitd.NewClient(0x1374048, 0xc0002fe840, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1aef9f8, ...)
	/earthly/buildkitd/buildkitd.go:83 +0x248
main.(*earthlyApp).actionBuildImp(0xc000132d80, 0xc00007f400, 0xc000320600, 0x0, 0x1, 0xc000320610, 0x1, 0x1, 0x0, 0x0)
	/earthly/cmd/earthly/main.go:2648 +0x32f
main.(*earthlyApp).actionBuild(0xc000132d80, 0xc00007f400, 0x1a4b280, 0xc0007aa000)
	/earthly/cmd/earthly/main.go:2581 +0x339
github.com/urfave/cli/v2.(*App).RunContext(0xc0004c3380, 0x1374048, 0xc0002fe840, 0xc00003c040, 0x2, 0x2, 0x0, 0x0)
	/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:322 +0x6fe
main.(*earthlyApp).run(0xc000132d80, 0x1374048, 0xc0002fe840, 0xc00003c040, 0x2, 0x2, 0x0)
	/earthly/cmd/earthly/main.go:1469 +0xad
main.main()
	/earthly/cmd/earthly/main.go:248 +0x485

$ DOCKER_BUILDKIT=1 docker build .
failed to dial gRPC: cannot connect to the Docker daemon. Is 'docker daemon' running on this host?: dial unix /var/run/docker.sock: connect: permission denied

$ sudo groupadd docker
groupadd: group 'docker' already exists

$ sudo usermod -aG docker $USER

$ newgrp docker 

$ DOCKER_BUILDKIT=1 docker build .
[+] Building 0.2s (2/2) FINISHED                                                                                                                                                                                   
 => [internal] load build definition from Dockerfile                                                                                                                                                          0.1s
 => => transferring dockerfile: 2B                                                                                                                                                                            0.0s
 => [internal] load .dockerignore                                                                                                                                                                             0.1s
 => => transferring context: 2B

$ earthly +format
           buildkitd | Found buildkit daemon as docker container (earthly-buildkitd)
```

--- a/buildkitd/buildkitd.go
+++ b/buildkitd/buildkitd.go
@@ -459,8 +459,14 @@ func Stop(ctx context.Context, containerName string, fe containerutil.ContainerF
 // IsStarted checks if the buildkitd container has been started.
 func IsStarted(ctx context.Context, containerName string, fe containerutil.ContainerFrontend) (bool, error) {
 	infos, err := fe.ContainerInfo(ctx, containerName)
+	if err != nil {
+		return false, err
+	}
 	containerInfo, ok := infos[containerName]
-	return err == nil && ok && containerInfo.Status == containerutil.StatusRunning, nil
+	if !ok {
+		return false, err
+	}
+	return containerInfo.Status == containerutil.StatusRunning, nil
 }
 
 // WaitUntilStarted waits until the buildkitd daemon has started and is healthy.

